File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
       ListNode res = new ListNode();
        ListNode cur = res;
        int c= 0;

        while(l1!=null || l2!=null || c!=0){
            int v1 = l1==null? 0:l1.val;
            int v2 = l2==null? 0:l2.val;
            int sum = v1+v2+c;
            cur.next = new ListNode(sum%10);
            c = sum/10;

            l1 = (l1!=null)? l1.next:null;
            l2 = (l2 != null) ? l2.next:null;
            cur=cur.next;
        }
        return res.next;
         
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def addTwoNumbers(self, l1, l2):
        carry=0
        r = n = ListNode(0)
        while l1 or l2 or carry:
            v1 = v2 =0
            if l1:
                v1 = l1.val
                l1 = l1.next
            if l2:
                v2 = l2.val
                l2 = l2.next
            carry,val = divmod(v1+v2+carry,10)
            n.next = ListNode(val)
            n = n.next
        return r.next
            

        